### PR TITLE
Create POST /v1/kitchen Endpoint

### PR DESCRIPTION
**Pull request by Naphrum**
Card:  https://github.com/orgs/HTMElephant/projects/2/views/1?pane=issue&itemId=47756433

I created kitchen in the back end and made a POST request to create a kitchen that allows you to also add kitchen users by email and assign them roles. 

The `WHERE email IN ($1:list)` in the SQL is way to allow multiple arguments from a list to be past into the query.

### CODE REVIEW

File: db/get_users_by_emails.sql [hunk 1]
```diff
@@ -0,0 +1,3 @@
+SELECT id, email
+FROM users
+WHERE email IN ($1:list)
```
Comment (jacobdman) [inline db/get_users_by_emails.sql:3]: I think this is exactly what @KyleD8n and I were looking for! Kyle maybe after all the other cards are done we can refactor to allow users to select multiple categories again? If that's even functionality we still want in the app that is..

File: routes/v1/kitchen/kitchen.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+      }
+    });
+
+    await db.kitchen_users.insert(
+      kitchenUsers.map((kitchenUser) => ({
+        kitchen_id: kitchen.id,
+        role: kitchenUser.role,
+        user_id: kitchenUser.id,
+      }))
+    );
```
Comment (jacobdman) [inline routes/v1/kitchen/kitchen.js:35]: This is complex, but very well done! Maybe let's insert a comment before the `forEach` to explain what it's doing?

```suggestion
    // iterate over the DB user records to match them with the new kitchen's user records
    matchedUsers.forEach((matchedUser) => {
      const result = kitchenUsers.findIndex((kitchenUser) => {
        return kitchenUser.email === matchedUser.email;
      });
      if (result >= 0) {
        kitchenUsers[result].id = matchedUser.id;
      }
    });

    await db.kitchen_users.insert(
      kitchenUsers.map((kitchenUser) => ({
        kitchen_id: kitchen.id,
        role: kitchenUser.role,
        user_id: kitchenUser.id,
      }))
    );
```

**Review by jacobdman — APPROVED** [review]
Excellent work Nate! I left a comment to leave a comment, but it's not a blocking/required change! I'm approving now, but if you decide to make the change I suggested I can approve again!

**Review by KyleD8n — COMMENTED** [review]
Amazing work Nathan

**Review by KyleD8n — APPROVED** [review]

**Pull request merged**